app/components/base_component.rb

```diff
@@ -8,7 +8,11 @@ def before_render
   end
 
   def self.scripts
-    @scripts ||= _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+    @scripts ||= begin
+      scripts = _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+      scripts.concat superclass.scripts if superclass != BaseComponent
+      scripts
+    end
```

Comment:
maybe this is a good reason to move this to be an instance method? it would be easier to call say `[...super, _sidecar_files]` etc? I notice all the places we load this call `self.class.scripts` so we already are on the instance level at that point?

Reply:
So maybe it would be

```
self.scripts # just this class's scripts

def scripts
  super + [self.class.scripts.map { ... }]
end
```

Reply:
I think part of the intent with the current implementation was to be able to memoize component class scripts persistently across requests, which wouldn't happen with instance variables, unless I'm mistaken (which could very well be the case).

Regardless, curious what would be the motivation for making it an instance method?

Reply:
FYI: Since this wasn't directly related to reCAPTCHA, I extracted a separate pull request for review at #7772.

Reply:
> Regardless, curious what would be the motivation for making it an instance method?

@-variables on classes and @-variables on instances behave differently so I just tend to avoid them on classes